MetaMask/metamask-mobile
# build: Pull `@metamask/mobile-provider` back into `metamask-mobile`

## **Description**

Currently the inpage injected scripts used in the in-app browser (for making the wallet provider object available to the dapp) are currently built in a separate `@metamask/mobile-provider` package. This PR brings the build process into the mobile repo itself. This is needed in order to support passing build time variables to the provider initialization (for EIP6963 support). 

## **Manual testing steps**

1. Pull in branch
2. `yarn setup`
3. `yarn watch`
4. `yarn start:ios`
5. Go to in-app browser
6. Visit test dapp
7. Verify that buttons are still functional
8. Optionally: Open safari debug console on the test dapp webview
9. check that `window.ethereum` is defined and `window.ethereum.chainId` logs a deprecation warning

## **Screenshots/Recordings**
Note the deprecation warnings which indicates that provider v13.0.0 is now being used
![image](https://github.com/MetaMask/metamask-mobile/assets/918701/143a5f56-29d4-4b7c-8192-d4595689ebba)


## **Related issues**

See: https://github.com/MetaMask/MetaMask-planning/issues/869

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
